### PR TITLE
RangeError and TypeError are JS exception; SyntaxError normally not

### PR DESCRIPTION
**Pull request by teoli2003**
Most exceptions in web/API are `DOMException`s.

But there are three special cases:
-  `RangeError` and `TypeError` are JS exceptions; 
- `SyntaxError` can be both but is in 99% of the cases a `DOMException` in web/API (the JS exception is reserved for the use of the JS parser).

This PR fixes this. It is larger than what I initially expected, a kind of a rabbit hole here. Sorry.

### CODE REVIEW

File: files/en-us/web/api/backgroundfetchmanager/fetch/index.md [hunk 1]
```diff
@@ -38,16 +38,12 @@ A {{jsxref("Promise")}} that resolves with a {{domxref("BackgroundFetchRegistrat
...
-- DOMException
-
-  - : This method may raise a {{domxref("DOMException")}} of the following types:
-
-    | Exception         | Description                                                                     |
-    | ----------------- | ------------------------------------------------------------------------------- |
-    | `AbortError`      | Indicates the fetch was aborted.                                                |
-    | `NotAllowedError` | Indicates that user permission has not been granted to make background fetches. |
+- `AbortError` {{domxref("DOMException")}}
+  - : Indicates the fetch was aborted.
```
Comment (wbamberg) [inline files/en-us/web/api/backgroundfetchmanager/fetch/index.md:44]:
```suggestion
  - : Indicates that the fetch was aborted.
```

File: files/en-us/web/api/cache/put/index.md [hunk 2]
```diff
@@ -61,8 +61,10 @@ put(request, response)
 
 A {{jsxref("Promise")}} that resolves with `undefined`.
 
-> **Note:** The promise will reject with a `TypeError` if the
-> URL scheme is not `http` or `https`.
+### Exceptions
+
+- {{jsxref("TypeError")}}
+  - : Returned if the URL scheme is not `http` or `https`.
```
Comment (wbamberg) [inline files/en-us/web/api/cache/put/index.md:67]: I feel like we had this conversation before about wording in cases where a promise is rejected with an exception, but I can't remember what we decided. Returned seems a bit weird but so does thrown.

File: files/en-us/web/api/contactsmanager/select/index.md [hunk 3]
```diff
@@ -62,11 +62,11 @@ Returns a {{jsxref('Promise')}} that resolves with an array of objects containin
 ### Exceptions
 
 - `InvalidStateError` {{domxref("DOMException")}}
-  - : Returned if the browsing context is not top-level or the contact picker is showing a flag. A flag denotes an already existing contact picker; only one picker can exist at any time.
+  - : Thrown if the browsing context is not top-level or the contact picker is showing a flag. A flag denotes an already existing contact picker; only one picker can exist at any time.
```
Comment (wbamberg) [inline files/en-us/web/api/contactsmanager/select/index.md:65]: ...and this is "Thrown" - or is the idea that this is actually thrown, and it doesn't reject a returned promise in this case?

File: files/en-us/web/api/cache/put/index.md (hunk 2 above)
Comment (teoli2003) [reply files/en-us/web/api/cache/put/index.md:67]: Yes, I avoided _Thrown_ here. We have to fix this at some point. We have a lot of _Returned_ for the moment (only for Promises), so we can find this one at the same time.

I can open an issue and make a proposal for these cases (and we should document it in the template)

File: files/en-us/web/api/contactsmanager/select/index.md (hunk 3 above)
Comment (teoli2003) [reply files/en-us/web/api/contactsmanager/select/index.md:65]: No it is an overcorrection. I shouldn't have.

File: files/en-us/web/api/cache/put/index.md (hunk 2 above)
Comment (wbamberg) [reply files/en-us/web/api/cache/put/index.md:67]: Yeah, maybe we should say something explicit like "The promise is rejected with...". But I'm fine not to deal with it in this PR :).

File: files/en-us/web/api/mediacapabilities/encodinginfo/index.md [hunk 4]
```diff
@@ -48,11 +48,11 @@ Browsers will report a supported media configuration as `smooth` and `powerEffic
...
-A `TypeError` is raised if the `MediaConfiguration` passed to the
-`encodingInfo()` method is invalid, either because the type is not video or
-audio, the `contentType` is not a valid codec MIME type, or any other error
-in the media configuration passed to the method, including omitting any of the [media encoding configuration](/en-US/docs/Web/API/MediaEncodingConfiguration)
-elements.
+- {{jsxref("TypeError")}}
+  - : Thrown if the `MediaConfiguration` passed to the `encodingInfo()` method is invalid,
+    either because the type is not video or audio, the `contentType` is not a valid codec MIME type,
+    or any other error in the media configuration passed to the method,
+    including omitting any of the [media encoding configuration](/en-US/docs/Web/API/MediaEncodingConfiguration) elements.
```
Comment (wbamberg) [inline files/en-us/web/api/mediacapabilities/encodinginfo/index.md:55]:
```suggestion
  - : Thrown if the `MediaConfiguration` passed to the `encodingInfo()` method is invalid, which may be for any of the following reasons:
    - the type is not video or audio
    - the `contentType` is not a valid codec MIME type,
    - there is some other error in the media configuration passed to the method, including omitting any of the [media encoding configuration](/en-US/docs/Web/API/MediaEncodingConfiguration) elements.
```

(or something. Let's make this long sentence easier to scan.)

File: files/en-us/web/api/merchantvalidationevent/complete/index.md [hunk 5]
```diff
@@ -42,7 +42,7 @@ complete(merchantSessionPromise)
 This exception may be passed into the rejection handler for the promise:
 
 - `InvalidStateError` {{domxref("DOMException")}}
-  - : Returned if the event did not come directly from the user agent, but was instead dispatched by other code. Another payment request is currently being processed, the current payment request is not currently being displayed to the user, or payment information is currently being updated.
+  - : Thrown if the event did not come directly from the user agent, but was instead dispatched by other code. Another payment request is currently being processed, the current payment request is not currently being displayed to the user, or payment information is currently being updated.
```
Comment (wbamberg) [inline files/en-us/web/api/merchantvalidationevent/complete/index.md:45]: Also "Thrown" for a promise.

File: files/en-us/web/api/rtcpeerconnection/addtransceiver/index.md [hunk 6]
```diff
@@ -50,8 +50,8 @@ addTransceiver(trackOrKind, init)
 
 ### Exceptions
 
-- `TypeError`
-  - : A string was specified as `trackOrKind` which is not valid. The string
+- {{jsxref("TypeError")}}
+  - : Thrown if an invalide string was specified as `trackOrKind`. The string
     must be either `"audio"` or `"video"`.
```
Comment (wbamberg) [inline files/en-us/web/api/rtcpeerconnection/addtransceiver/index.md:55]:
```suggestion
- {{jsxref("TypeError")}}
  - : Thrown if `trackOrKind` was not either `"audio"` or `"video"`.
```

File: files/en-us/web/api/xrray/xrray/index.md [hunk 7]
```diff
@@ -37,11 +37,11 @@ A newly-created {{domxref("XRRay")}} object.
...
-A `TypeError` is thrown,
-
-- if all of `direction`'s `x`, `y`, and `z` coordinates are zero.
-- if `direction`'s `w` coordinate is not 0.0.
-- if `origin`'s `w` coordinate is not 1.0.
+- {{jsxref("TypeError")}}
+  - : Thrown if one of the following conditions is met:
+    - if all of `direction`'s `x`, `y`, and `z` coordinates are zero.
+    - if `direction`'s `w` coordinate is not 0.0.
+    - if `origin`'s `w` coordinate is not 1.0.
```
Comment (wbamberg) [inline files/en-us/web/api/xrray/xrray/index.md:44]:
```suggestion
    - all of `direction`'s `x`, `y`, and `z` coordinates are zero.
    - `direction`'s `w` coordinate is not 0.0.
    - `origin`'s `w` coordinate is not 1.0.
```

**Comment by wbamberg** [comment]
All done. Sorry for the "single comment" way I did this.

File: files/en-us/web/api/cache/put/index.md (hunk 2 above)
Comment (teoli2003) [reply files/en-us/web/api/cache/put/index.md:67]: I made a note to open and lead the discussion here. I need to dig a bit to make an educated first proposal.

**Review by wbamberg — APPROVED** [review]
👍 nice work!

**Pull request merged**